### ISSUE TITLE
GUI getResourceAsStream not working -- Null pointer exception

### ISSUE THREAD

**Issue by whtan99** [nus-cs2103-AY2122S2/forum#76]
Hi everyone, I'm on part-3 of the tutorial for building GUI.

In the tutorial, they asked us to input these two lines:
```java
private Image user = new Image(this.getClass().getResourceAsStream("/images/User.jpg"));
private Image duke = new Image(this.getClass().getResourceAsStream("/images/Duke.jpg"));
```

However, when I try to build/run this, I get this error:
![Screenshot 2022-02-01 at 11 12 54 PM](https://user-images.githubusercontent.com/54658569/151995536-2a3aa967-ccc3-472f-a4a7-d7610c637222.png)

My directory, for reference:
![Screenshot 2022-02-01 at 11 13 03 PM](https://user-images.githubusercontent.com/54658569/151995618-12bdbefb-f483-4019-8cfa-e6148b2246b7.png)

Does anyone know a fix to this? Thanks!

**Comment by takufunkai** [comment]
Hi Wei Howe, did you [try rebuilding your project?](https://stackoverflow.com/a/57218503)

**Comment by damithc** [comment]
@goalfix Your error message has `main.java.duke.Duke`. Shouldn't that be `duke.Duke`? `main.java` are not supposed to be packages.

**Comment by whtan99** [comment]
Hi @takufunkai, thanks for your suggestion. I have tried rebuilding the project but to no avail.

Hi prof, @damithc, I have not been able to put packages solely in `Duke`. For reference, this is my current project structure.
![Screenshot 2022-02-02 at 3 17 15 PM](https://user-images.githubusercontent.com/54658569/152110435-72e6ca8f-35f0-4f3b-82d3-3aaed29c46bb.png)

If I attempt to change the package to duke, as shown here, intellij gives me a error, saying that it does not correspond to the file path `main.java.duke`.

![Screenshot 2022-02-02 at 3 18 26 PM](https://user-images.githubusercontent.com/54658569/152110592-3ae57e2e-2d44-4969-bd04-b39bc6891390.png)

This is strange, as I'm able to put test classes (which are found in `src/test/java/duke/`) into package duke, but not `src/main/java/duke/` into package duke.

Any suggestions would be welcome! Thanks

**Comment by whtan99** [comment]
Also, I was able to finish up till part 2 of the GUI tutorial! Just seems the loading of images has some issues 😅

**Comment by damithc** [comment]
> Hi prof, @damithc, I have not been able to put packages solely in `Duke`. 
This is a problem you need to fix first. `main.java` is not meant to be packages. I assume you are using Gradle already? If not, it's best to start using Gradle before JavaFX. It might fix this problem as well.

**Comment by whtan99** [comment]
Hi, I have resolved this issue. 
>have not been able to put packages solely in Duke

I fixed this, by adding `IP` as a module. Originally, only `src/main` was a module. So I believe the project should have `IP`, `Main` and `Test` as modules.

>could not load images in

It turned out that the file path was referencing my JDK `images" folder. By changing the file in which I put the image in as a resource folder (going to project structure -> modules -> resource folder) instead of a normal folder, this issue was resolved. 

I also reinitialised gradle, so there could have been some errors there. My project structure now, for reference:
![Screenshot 2022-02-05 at 5 50 29 PM](https://user-images.githubusercontent.com/54658569/152636830-70d61099-219e-4376-a85d-0d08a85b6a4b.png)



Thanks all! @damithc @takufunkai
